src/frame/stream_id.rs

```diff
@@ -25,25 +37,34 @@ impl StreamId {
         (StreamId(unpacked & !STREAM_ID_MASK), flag)
     }
 
+    /// Returns true if this stream ID corresponds to a stream that
+    /// was initiated by the client.
     pub fn is_client_initiated(&self) -> bool {
         let id = self.0;
         id != 0 && id % 2 == 1
     }
 
+    /// Returns true if this stream ID corresponds to a stream that
+    /// was initiated by the server.
     pub fn is_server_initiated(&self) -> bool {
         let id = self.0;
         id != 0 && id % 2 == 0
     }
 
+    /// Return a new `StreamId` for stream 0.
     #[inline]
     pub fn zero() -> StreamId {
         StreamId::ZERO
     }
 
+    /// Returns true if this stream ID is zero.
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
 
+    /// Returns the next stream ID initiated by the same peer as this stream
+    /// ID, or an error if incrementing this stream ID would overflow the
+    /// maximum.
     pub fn next_id(&self) -> Result<StreamId, StreamIdOverflow> {
```

Comment:
Again, I'm not sure whether or not we want users of the library to be able to construct their own `StreamId`s. It seems harmless but I'm not sure if there's any use-case for it. Perhaps this should be `pub(crate)`?